watertap/unit_models/ion_exchange_0D.py

```diff
@@ -1053,7 +1054,7 @@ def eq_partition_ratio(b):
                 self.target_ion_set, doc="Removed total mass of ion in equivalents"
             )
             def eq_mass_removed(b, j):
-                charge = prop_in.charge_comp[j]
+                charge = abs(value(prop_in.charge_comp[j]))
```

Comment:
```suggestion
                charge = abs(prop_in.charge_comp[j])
```
This should work just fine as it is a parameter. Pyomo will create an expression for it such that even if the parameter value changes you will get the correct result.

If `charge_comp` has units you might need to remove and re-apply them after outside the `abs` function.